### PR TITLE
Port platform_info table to M1 Macs

### PR DESCRIPTION
**Pull request by josephsweeney**
Add port for the `platform_info` for M1 Macs, where the information is in different locations in the IODeviceTree registry.

The following columns appear to not have a equivalent data on M1 Macs:

- `vendor`
- `volume_size`
- `size`
- `date`
- `version`

### CODE REVIEW

File: osquery/tables/system/darwin/smbios_tables.cpp [hunk 1]
```diff
@@ -281,5 +282,62 @@ QueryData genPlatformInfo(QueryContext& context) {
...
+  Row r;
+  r["vendor"] = getIOKitProperty(details, "manufacturer");
+
+  auto chosen =
+      IORegistryEntryFromPath(kIOMasterPortDefault, "IODeviceTree:/chosen");
+  if (chosen != 0) {
+    IORegistryEntryCreateCFProperties(
+        chosen, &details, kCFAllocatorDefault, kNilOptions);
+    IOObjectRelease(chosen);
+    r["version"] = getIOKitProperty(details, "system-firmware-version");
```
Comment (directionless) [inline osquery/tables/system/darwin/smbios_tables.cpp:310]: I don't know what correct is, but on my m1 machine this PR return `iBoot-7459.121.3` which seems a little weird? And I notice that's not what system profiler says:
```
$ system_profiler SPHardwareDataType | grep -i "Version"
      System Firmware Version: 7459.121.3
      OS Loader Version: 7459.121.3
```

**Comment by josephsweeney** [comment]
After investigation, the PR turns out to be fully functional. Could we remove the "do not merge" tag?

File: osquery/tables/system/darwin/smbios_tables.cpp (hunk 1 above)
Comment (directionless) [reply osquery/tables/system/darwin/smbios_tables.cpp:310]: I feel like we should strip `iBoot` but I could be convinced it's not worth it, and to see how it does in the broader world first

Comment (alessandrogario) [reply osquery/tables/system/darwin/smbios_tables.cpp:310]: I think the System Information.app application is trying to report less information to the user to make things simpler to understand, but that is definitely an exception.

I personally prefer the output as-is, which is what any other tool will report (such as ioreg) by just reading and returning the actual data without modifications

**Pull request merged**